app/tasks/job_attributes_fetcher.rb

```diff
@@ -39,9 +39,9 @@ def application_question_set
   end
 
   def apply_with_cheddar
-    return false unless application_question_set&.first&.dig(:questions)
+    return false unless application_question_set.dig(:core_questions, :questions)
 
-    application_question_set.first[:questions]
+    application_question_set.dig(:core_questions, :questions)
                             .none? { |question| apply_with_cheddar_conditions(question) }
```

Comment:
Can we add comments to this file please for what it does

Reply:
Also do we need that many variables in the initialise method?

Reply:
I did  not work much on that file. ... I don't know.